### PR TITLE
Added Cocktail Sort in c

### LINKED ISSUE #538: Cocktail sort  in c++
I would like to on with it. (it is a variation of Bubble sort)

### PR DESCRIPTION
**Pull request by somya-kapoor**
Fixes #538 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Sample Input / Output.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

#### Changes proposed in this pull request:

- Added Cocktail Sort in c

#### Languages Used:
- C

#### Files Added:
- Cocktail_Sort.c

### CODE REVIEW

File: Cocktail_Sort/Cocktail_Sort.c [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    {  
+        swapped = 0; 
+        // iterate through every element from left to right
+        for (i = begin; i < end; ++i) 
+        {    
+            // Move the bigger element to the right most position as in bubble sort
+            if (a[i] > a[i + 1])
+            {  
+                temp = a[i];  
+                a[i]=a[i+1];
```
Comment (singh-shreya6) [inline Cocktail_Sort/Cocktail_Sort.c]: space before and after equal to sign at all places

**Comment by singh-shreya6** [comment]
The checkbox in PR template can be ticked by putting x inside square brackets like this [x]. Please add a brief introduction about algorithm at the top of code.

File: Cocktail_Sort/Cocktail_Sort.c [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+        // if nothing moved, then array is sorted. 
+        if (!swapped)  
+            break;  
+        swapped = 0; 
+        // moving the smaller elements to left part of the array which is modification over bubble sort
+        for (i = end - 1; i >= begin; --i)
+        {  
+            if (a[i] > a[i + 1])   
+            {
+                temp = a[i];
```
Comment (jainaman224) [inline Cocktail_Sort/Cocktail_Sort.c]: Creation of swap function would be better as it is used twice.

File: Cocktail_Sort/Cocktail_Sort.c [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+//algorithm and a comparison sort. The algorithm differs from a bubble sort in 
+//that it sorts in both directions on each pass through the list. 
+#include <stdio.h>  
+   
+int main()  
+{  
+    int n;
+    printf("Enter the number of elements ");
+    scanf("%d",&n);
+    int a[n];
```
Comment (jainaman224) [inline Cocktail_Sort/Cocktail_Sort.c]: Use of better variable names than `a` and `n` will be amazing.

Comment (somya-kapoor) [reply Cocktail_Sort/Cocktail_Sort.c]: @jainaman224  I have done the required changes, please review.

**Pull request merged**